ehitchhiking-ui/src/app/pages/main-screen/yandex-map/yandex-map.component.ts

```diff
@@ -181,10 +182,24 @@ export class YandexMapComponent implements OnInit, OnChanges {
 			.catch((error) => console.log('Failed to load Yandex Maps', error));
 	}
 
-	setInfoAboutRoute(multiRoute, data: Partial<Route>) {
+  setInfoAboutRoute(multiRoute, index: number) {
 		multiRoute.events.add('update', (e) => {
 			const activeRoute = multiRoute.getActiveRoute();
-			data.tripDuration = activeRoute.properties.get('duration').text;
+      this.routes[index].tripDuration = activeRoute.properties.get('duration').text;
+
+      const coordinates = multiRoute.model.getReferencePoints();
```

Comment:
const coordinates = multiRoute.model.getReferencePoints();
      this.ymapsPromise
        .then((maps) => {
          maps.geocode(coordinates[0]).then((res) => {
            const firstGeoObject = res.geoObjects.get(0);
            this.routes[index].from = firstGeoObject.getAddressLine();
          });
          maps.geocode(coordinates[1]).then((res) => {
            const firstGeoObject = res.geoObjects.get(0);
            this.routes[index].to = firstGeoObject.getAddressLine();
          });
        });

move to separate function

Reply:
fixed